src/DataSource.ts

```diff
@@ -1,12 +1,25 @@
-import { DataSourceInstanceSettings, MetricFindValue } from '@grafana/data';
+import { DataSourceInstanceSettings, MetricFindValue, DataQueryRequest, DataQueryResponse } from '@grafana/data';
 import { DataSourceWithBackend, getTemplateSrv } from '@grafana/runtime';
 import { InfluxQuery, InfluxOptions } from './types';
+import { Observable } from 'rxjs';
 
 export class DataSource extends DataSourceWithBackend<InfluxQuery, InfluxOptions> {
-  constructor(instanceSettings: DataSourceInstanceSettings<InfluxOptions>) {
+  constructor(private instanceSettings: DataSourceInstanceSettings<InfluxOptions>) {
     super(instanceSettings);
   }
 
+  query(request: DataQueryRequest<InfluxQuery>): Observable<DataQueryResponse> {
+    // Why? This is needed otherwise options never get passed into the query (ex: v.bucket)
```

Comment:
?????  this does not make any sense!  this will attach the instance settings to every query and have it saved in the panel.json  You should be able to get bucket from:

https://github.com/grafana/influxdb-flux-datasource/blob/master/pkg/models/query.go#L29

Or if you need instance settings:
https://github.com/grafana/influxdb-flux-datasource/blob/master/pkg/models/settings.go#L32